### PR TITLE
cpu-o3: add perfCCT frame, add inst lifetime counter

### PR DESCRIPTION
**Pull request by tastynoob**

### CODE REVIEW

File: src/cpu/o3/commit.cc [hunk 1]
```diff
@@ -1676,14 +1678,6 @@ Commit::commitHead(const DynInstPtr &head_inst, unsigned inst_num)
     // Finally clear the head ROB entry.
     rob->retireHead(tid);
 
-#if TRACING_ON
-    if (debug::O3PipeView) {
-        head_inst->commitTick = curTick() - head_inst->fetchTick;
-        DPRINTF(O3PipeView, "Record commit for inst sn:%lu, commitTick=%lu\n",
```
Comment (jueshiwenli) [inline src/cpu/o3/commit.cc:1682]: It is better to keep o3pipview.

File: src/cpu/o3/perfCCT.hh [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        AtRename,
+        AtDispQue,
+        AtIssueQue,
+        AtIssueArb,
+        AtIssueReadReg,
+        AtFU,
+        AtBypassVal,
+        AtWriteVal,
+        AtCommit,
+        InvalidPos
```
Comment (shinezyy) [inline src/cpu/o3/perfCCT.hh]: Can we define this in Python's enumeration and use it when create SLQ table to avoid redundant code?

**Review by shinezyy — CHANGES_REQUESTED** [review]

**Review by jueshiwenli — APPROVED** [review]

**Review by jueshiwenli — APPROVED** [review]

**Review by shinezyy — APPROVED** [review]
Redudant codes are now removed, excellent!

**Comment by dzwduan** [comment]
Why not consider using https://github.com/shioyadan/Konata for pipeline visualization ?

**Pull request merged**